cmd/kube-apiserver/app/server.go

```diff
@@ -115,6 +116,9 @@ func Run(s *options.ServerRunOptions) error {
 	if err = s.Authentication.Apply(genericConfig); err != nil {
 		return fmt.Errorf("failed to configure authentication: %s", err)
 	}
+	if err := preflight.RunApiserverChecks(s); err != nil {
```

Comment:
Alright, now making it all the way to usage, I think I'd like `utilwait.PollImmediate(interval, duration, conditionfunc)` where you provide the condition func for testing etcd servers as a function on an struct.


```go
err := utilwait.PollImmediate(100 milliseconds, 10 seconds, NewEtcdCheck(serverlist).Condition)
if err != nil{
    return err
}
```